src/compiler/parser.ts

```diff
@@ -2054,16 +2054,17 @@ namespace ts {
             parsingContext |= 1 << kind;
             const list = [];
             const listPos = getNodePos();
+            let lastElement: Node | undefined;

```

Comment:
why do you need this?  is this not just `list[list.length - 1]`?  (i think TS may have a `last` helper as well).

Reply:
Because it's not necessarily the last statement in the list.

```ts
{ hello } = foo();
yadda()
```

Reply:
i would def be a bit concerned here.  and i wouldn't do it htis way anyways.  Instead, just have the block parsing code check for an errant `=` that follows it.  It can then say: whoops, person may have intended to do a destructuring assignment, report proper error there and consume the `=`.  then you'll be back in a good state, and you dn't have to touch the core parsing loop.  it also puts the error right where it is initially seen.  not in the recovery logic.